Description: Work around for GNU/Hurd
 Newer glibc hide the definition of JB_SP/JB_PC as private,
 so, there is no clean way other than define them here.
 GNU/Linux or GNU/kfreebsd doesn't use this version of pth_mctx_set
 anymore (because their mctx implementation now is mcsc/sc/mc),
 so, we have no problem modifying this.
Author: NIIBE Yutaka <gniibe@fsij.org>
Bug-Debian: http://bugs.debian.org/466115
Forwarded: no
Reviewed-By: NIIBE Yutaka <gniibe@fsij.org>
Last-Update: 2010-05-12

Index: pth-2.0.7/pth_mctx.c
===================================================================
--- pth-2.0.7.orig/pth_mctx.c	2013-07-08 08:33:26.000000000 +0000
+++ pth-2.0.7/pth_mctx.c	2013-07-08 08:33:30.000000000 +0000
@@ -465,6 +465,10 @@
     pth_mctx_t *mctx, void (*func)(void), char *sk_addr_lo, char *sk_addr_hi)
 {
     pth_mctx_save(mctx);
+#if defined(__i386__) /* Workaround for GNU/Hurd */
+#define JB_SP	4
+#define JB_PC	5
+#endif
 #if defined(__GLIBC__) && defined(__GLIBC_MINOR__) \
     && __GLIBC__ >= 2 && __GLIBC_MINOR__ >= 0 && defined(JB_PC) && defined(JB_SP)
     mctx->jb[0].__jmpbuf[JB_PC] = (int)func;
